### PR TITLE
Andersen pta

### PR DESCRIPTION
**Pull request by e-avidan**

### CODE REVIEW

File: data/typescript/lib/net.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { EventEmitter } from './events';
 class Server extends EventEmitter {
     _handle: TCP
 
-    private constructor(optionsOrCallback: ServerOptions | ((c: Socket) => void),
+    private constructor(§optionsOrCallback: ServerOptions | ((c: Socket) => void),
```
Comment (corwin-of-amber) [inline data/typescript/lib/net.ts:11]: What this?

File: data/typescript/lib/net.ts [hunk 2]
```diff
@@ -20,8 +20,29 @@ class Server extends EventEmitter {
...
+        // var x = interesting2();
+
         try {
             h.bind(port); h.listen(5);
+
+            // And to make them even more interesting :-)
+            z = h;
+            if (z == 42) {
+                return 15;
+            }
```
Comment (corwin-of-amber) [inline data/typescript/lib/net.ts:45]: I suggest we put these snippets into a separate test file, and then we can make it into a small test suite.

File: src/analysis/syntax.ts [hunk 3]
```diff
@@ -1,14 +1,71 @@
...
+export const RETURN_STATEMENT = "ReturnStatement";
+
+export const CALL_EXPRESSION = "CallExpression";
+export const BINARY_EXPRESSION = "BinaryExpression";
+export const ARRAY_LITERAL_EXPRESSION = "ArrayLiteralExpression";
+export const OBJECT_LITERAL_EXPRESSION = "ObjectLiteralExpression";
+export const TYPE_ASSERTION_EXPRESSION = "TypeAssertionExpression";
+export const PROPERTY_ACCESS_EXPRESSION = "PropertyAccessExpression";
+export const NEW_EXPRESSION = "NewExpression";
+export const PARANTHESIZED_EXPRESSION = "ParenthesizedExpression";
```
Comment (corwin-of-amber) [inline src/analysis/syntax.ts:23]: This is extremely meticulous.

I would argue though that the AST node types are in effect an enum, hence creating a shadow constant for each value is a bit much. If you are worried about typos, a single union type can help:
```typescript
type ASTNodeType = "ClassDeclaration" | "MethodDeclaration" | ...
```

File: data/typescript/lib/net.ts (hunk 1 above)
Comment (e-avidan) [reply data/typescript/lib/net.ts:11]: bad characters got it, will delete

File: data/typescript/lib/net.ts (hunk 2 above)
Comment (e-avidan) [reply data/typescript/lib/net.ts:45]: ok, I'll create a separate file

File: src/analysis/syntax.ts (hunk 3 above)
Comment (e-avidan) [reply src/analysis/syntax.ts:23]: Less about typos, more about me having to reference the types when parsing the code
I'd tried to minimize it as much as possible, but in order to build the PTA graph I had to incorporate some language knowledge

Comment (e-avidan) [reply src/analysis/syntax.ts:23]: Actual strings in switch case / no need for them here

**Pull request merged**